### PR TITLE
[SPARK-30834][DOCS][PYTHON] Add note for recommended pandas and pyarrow versions

### PR DESCRIPTION
**Pull request by BryanCutler**
### What changes were proposed in this pull request?

Add doc for recommended pandas and pyarrow versions. 

### Why are the changes needed?

The recommended versions are those that have been thoroughly tested by Spark CI. Other versions may be used at the discretion of the user.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

NA

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @HyukjinKwon @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
Thanks, @BryanCutler . It looks good for `master` and `3.0`.

**Comment by holdenk** [comment]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

File: docs/sql-pyspark-pandas-with-arrow.md [hunk 1]
```diff
@@ -255,6 +255,12 @@ different than a Pandas timestamp. It is recommended to use Pandas time series f
 working with timestamps in `pandas_udf`s to get the best performance, see
 [here](https://pandas.pydata.org/pandas-docs/stable/timeseries.html) for details.
 
+### Recommended Pandas and PyArrow Versions
+
+For usage with pyspark.sql, the supported versions of Pandas is 0.24.2 and PyArrow is 0.15.1. Higher
```
Comment (HyukjinKwon) [inline docs/sql-pyspark-pandas-with-arrow.md]: @BryanCutler, seems we should fix one instance at "Ensure PyArrow Installed" too.

Comment (HyukjinKwon) [inline docs/sql-pyspark-pandas-with-arrow.md]: Bryan, we should maybe fix 0.15+ to just 0.15.1 at "Ensure PyArrow Installed".

**Comment by HyukjinKwon** [comment]
I just addressed my minor comment. I manually built and checked the documentation as well. We don't build the doc in PR builders. Let me merge this.

Merged to master.

**Comment by BryanCutler** [comment]
Thanks @dongjoon-hyun and @holdenk !

**Pull request closed without merging**